ark/api/v1alpha1/query_types.go

```diff
@@ -138,7 +138,19 @@ func (q *QuerySpec) GetInputMessages() ([]openai.ChatCompletionMessageParamUnion
 	}
 
 	var messages []openai.ChatCompletionMessageParamUnion
-	if err := json.Unmarshal(q.Input.Raw, &messages); err != nil {
+
+	// First try to unmarshal directly as array (for backward compatibility)
```

Comment:
I'm not sure what this does but should we need to change this function just to handle the memory changes?